optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/stream/AbstractConstraintStreamScoreDirectorFactory.java

```diff
@@ -65,21 +63,6 @@ protected AbstractConstraintStreamScoreDirectorFactory(SolutionDescriptor<Soluti
                 (ConstraintSessionFactory<Solution_, Score_>) constraintFactory.buildSessionFactory(constraints);
```

Comment:
This line should now happen in the drools/bavet subclasses. Those should also no longer doe the fancy supplier trick, just the validation of the constraints can go through a common method.